src/transforms/pipelines/config.rs

```diff
@@ -77,7 +77,7 @@ impl PipelineConfig {
             );
             result.outputs.push((
                 filter_name.clone(),
-                vec![Output::from((UNMATCHED_ROUTE, DataType::all()))],
+                vec![Output::default(DataType::all()).with_port(UNMATCHED_ROUTE)],
```

Comment:
I don't love `Output::default(...).with_port(...)` because the `default` there is intended to refer to a "default" output which is one without a port. That terminology is seeming more and more like a poor choice. 

Maybe `Output::named(port, type)` would be better? There's more we could do with a full builder implementation, but I'm not sure how worthwhile that'd be.

Reply:
I agree that it's awkward. I _did_ consider doing something like `Output::trace().with_port(...)` etc. So, there'd be one constructor per `DataType`, in addition to an `::all()`.

Reply:
I also experimented some, after this conversation, with `Output::named`. I think my basic concern there is, like the coercion from `(String, DataType)`, it's not really clear from just looking at it what `Output` _is_ where, with the more, uh, chatty building interface I think that, to my eyes, it's more apparent what this thing is doing.

Reply:
I guess I'm not sure where the `From` impl comes into play with regards to creating a new `Output::named` method. It feels objectively better, in terms of clarity of intent, to have a true method instead of the `From` conversion pathway.

Separately, but tangentially: maybe we should also have an `Output::unmatched` and globally reserve that `_unmatched` identifier? There's only a few fixed/reserved output names, so I don't think it hurts us to codify those.

Reply:
> I agree that it's awkward. I did consider doing something like Output::trace().with_port(...) etc. So, there'd be one constructor per DataType, in addition to an ::all().

One thing I had not considered is that `DataType` is a flag set, so this idea doesn't work. 

> Separately, but tangentially: maybe we should also have an Output::unmatched and globally reserve that _unmatched identifier? There's only a few fixed/reserved output names, so I don't think it hurts us to codify those.

I like this notion. 

I think, instead of blowing this PR out any further, I'll let the awkwardness alone here and leave tidying this up for future generations since I 1. like what we have got here so far and 2. am not _totally_ sure what I'd like to see next.